tests/parshaping-simple.sil

```diff
@@ -0,0 +1,89 @@
+local circle = {
+  { width = SILE.measurement("36.779999477581pt"), indent=SILE.measurement("149.44200026121pt") },
+  { width = SILE.measurement("89.148920336526pt"), indent=SILE.measurement("123.25753983174pt") },
+  { width = SILE.measurement("116.65835432049pt"), indent=SILE.measurement("109.50282283975pt") },
+  { width = SILE.measurement("135.41673924727pt"), indent=SILE.measurement("100.12363037636pt") },
+  { width = SILE.measurement("148.77195603998pt"), indent=SILE.measurement("93.446021980012pt") },
+  { width = SILE.measurement("158.09926166671pt"), indent=SILE.measurement("88.782369166646pt") },
+  { width = SILE.measurement("164.08698356407pt"), indent=SILE.measurement("85.788508217965pt") },
+  { width = SILE.measurement("167.09452358097pt"), indent=SILE.measurement("84.284738209513pt") },
+  { width = SILE.measurement("167.28269918420pt"), indent=SILE.measurement("84.190650407902pt") },
+  { width = SILE.measurement("164.66117660805pt"), indent=SILE.measurement("85.501411695975pt") },
+  { width = SILE.measurement("159.09112080046pt"), indent=SILE.measurement("88.286439599772pt") },
+  { width = SILE.measurement("150.24495450145pt"), indent=SILE.measurement("92.709522749275pt") },
+  { width = SILE.measurement("137.49177425846pt"), indent=SILE.measurement("99.086112870769pt") },
+  { width = SILE.measurement("119.58808312009pt"), indent=SILE.measurement("108.03795843996pt") },
+  { width = SILE.measurement("93.624309129279pt"), indent=SILE.measurement("121.01984543536pt") },
+  { width = SILE.measurement("47.922780548840pt"), indent=SILE.measurement("143.87060972558pt") },
+  { width = SILE.measurement("167.54343340503pt"), indent=SILE.measurement("84.060283297483pt") }
+}
+
+SILE.registerCommand("testparshaping", function (options, content)
+  local oldSetup = SILE.linebreak.setupLineLengths
+
+  -- Ship out previous content and start a new paragraph.
+  SILE.typesetter:leaveHmode()
+
+  SILE.settings.temporarily(function()
+    -- Ensure reasonably fixed baselineskip, and some more tolerance
+    -- as circles are hard.
+    SILE.settings.set("linebreak.tolerance", 2000)
+    SILE.settings.set("document.baselineskip", "1.2em")
+    SILE.settings.set("document.parindent", 0)
+
+    -- Override linebreak setup to implement our parshape
+    SILE.linebreak.setupLineLengths = function (self)    
+        self.parShape = function (self, line)
+            if circle[line] then
+                return {
+                    width = circle[line].width,
+                    left = circle[line].indent:tonumber(),
+                    right = circle[line].indent:tonumber()
+                }
+            end
+            -- Paragraphs that extend beyond the circle...
+            return {
+                width = self.hsize,
+                left = 0,
+                right = 0
+            }
+        end 
+        -- Assume we should never reach 500 lines. If it does, stop
+        -- the mess.
+        self.lastSpecialLine = 500
```

Comment:
I don't like this magic number. Bibles may have many small lines…please try to figure out a termination condition that isn't just "no one will ever want more than 500 lines", it's too Bill Gates and his 640KB of RAM.

Reply:
Simple here: no termination condition and trust parShape to always spit out values. Problem is that the line breaking algorithm expects a value here, and I frown upon messing too much with it.

Reply:
Done

Reply:
There are a pile of hard coded values in the break and shape mechanisms already, many of them inherited from TeX history. I've been meaning to abstract them to a known place in SILE memory that *could be* tuned, but that doesn't belong in this PR.